src/node.h

```diff
@@ -234,9 +257,24 @@ class NODE_EXTERN MultiIsolatePlatform : public v8::Platform {
   virtual void UnregisterIsolate(v8::Isolate* isolate) = 0;
 };
 
+// Set up some Node.js-specific defaults for `params`, in particular
+// the ArrayBuffer::Allocator if it is provided, memory limits, and
+// possibly a code event handler.
+NODE_EXTERN void SetIsolateCreateParams(v8::Isolate::CreateParams* params,
```

Comment:
Maybe it would make sense to have a subclass of `Isolate` (`NodeIsolate`?) Then this could be a static function and `SetIsolateUpForNode` could be something like `NodeIsolate::Initialize`, and we could enforce some type checks instead of having a loose idea of "an isolate with Node.js specific tweaks"

Reply:
I think that would run against the intention of this PR – It’s intentional that you can use an Isolate that has not had `SetIsolateUpForNode()` called on it with Node.js classes. It’s just that in that case the embedder has to control things like error handling itself.

(I’m not sure how well that would work right now, but I’d have that as a possible goal for embedding Node.js. https://github.com/nodejs/node/pull/21341 also kind of went into that direction)